aiconfig-docs/docs/usage-guide.md

```diff
@@ -23,46 +23,138 @@ import constants from '@site/core/tabConstants';
+
+Take this sample api call to openai:
+```python
+openai.Chat.completions.create(**{
+ 'top_p': 1,
+ 'temperature': 1,
+ 'model': 'gpt-3.5-turbo',
+ 'messages': [{'content': 'Tell me 10 fun attractions to do in NYC.','role': 'user'}]
+})
+```
+We can convert this into a prompt using the serialize method
+```python
+data = {
+ 'top_p': 1,
+ 'temperature': 1,
+ 'model': 'gpt-3.5-turbo',
+ 'messages': [{'content': 'Tell me 10 fun attractions to do in NYC.','role': 'user'}]
+}
+new_prompt = config.serialize("gpt-3.5-turbo",data, prompt_name="get_activities")
+config.add_prompt(new_prompt.name, new_prompt)
+```
+```typescript
+```
+
+
 ### Run a prompt
+You can execute a single prompt using your AIConfig anytime after you've loaded your config into memory. This allows you to obtain a generated response from the model.
+
+```python
+# Example code to run a prompt
+response = aiconfig_runtime.run(prompt_name="my_prompt")
+print(response)
+```
+Sample Output: 
+```
+Execute Result(
+  yada yada yada
+)
+```
+
+AIConfig will automatically store the response in a standardized format, namely `ExecuteResult`
+
+#### Outputs
+
+On Run, a model parser will store the response in a standardized format, namely `ExecuteResult`. This is a dictionary like object with the following attributes:
+- output_type
+- execution_count
+- data
+- mime_type: Optional[str] = None
+- metadata: Dict[str, Any]
+
+#### Retrieving Outputs
+
+A model parser has the option to implement method `get_output_text`. This method takes in a prompt name and optionally an output and returns the raw text response, unpacked from the Output format. If no output is specified, the last output should be returned.
+
+To call this method from the AIConfig Runtime, simply call `get_output_text` on the runtime object.
+
+```python
+output_text = AIConfig.get_output_text(prompt_name="my_prompt")
+```
 
 #### Run a prompt chain
 
-### Passing data into a prompt template
+Check out the [prompt chaining](#prompt-chaining) section for more details.
 
-### Serialize and deserialize a prompt
+### Passing data into a prompt template
 
-### Setting metadata
+Parameters! AIConfig supports handlebars style syntax for prompt templates. Check out the []
 
 ### Callbacks
 
+Callback
+
 #### Stream callbacks
 
-#### Event callbacks
+Some model parsers allow you to stream the output of a prompt as it is generated. This is useful for long running prompts, or prompts that generate a lot of output. In order to use this feature, you must specify a callback function when you run the prompt.
 
-## Creating an `aiconfig`
+#### Add Prompt
+There are a couple ways to dyanmically add prompts to a config at runtime. The first is to construct a Prompt object.
 
-### Programmatically
+```python
+new_prompt = Prompt(name="my_prompt", input="Hello! What is the difference between an apple and a banana")
+```
+```typescript
+const newPrompt = {name: "my prompt", input: "Hello! What is the difference between an apple and a banana?" }
```

Comment:
Hmmm, is this actually helpful to document? Isn't this an anti-pattern and we want them using `serialize`?